applications/ocelloids_xcm_monitoring_service.md

```diff
@@ -0,0 +1,270 @@
+# Ocelloids XCM Transfer Monitoring Service
+
+- **Team Name:** SO/DA zone
+- **Payment Address:** Fiat 04.05.2023, 16:37
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+## Project Overview
+
+This grant proposal is a follow-up to the Ocelloids Monitoring SDK, previously delivered and available here: https://github.com/w3f/Grant-Milestone-Delivery/pull/934
+
+The objective of this grant is to develop an open-source monitoring service using the Ocelloids SDK. This service will monitor XCM transfers across Asset Hub and selected parachains. The primary purpose is to offer exchanges, custodians and other institutions on Asset Hub an efficient means of tracking asset transfers across different parachains. Additionally, we will abstract the underlying pallets responsible for initiating XCM transfers and managing assets, which varies for each parachain. Users will have access to an HTTP API to subscribe to XCM transfers and manage their subscriptions. A Docker image will be provided to facilitate service deployment.
+
+## Project Details
+
+This service is designed for use alongside an Asset Hub node deployment and the Asset Transfer API. The goal is to provide exchanges, custodians and institutions with an easily deployable and manageable monitoring solution for tracking their asset flows across parachains. The service will also be available for any user interested in monitoring XCM transfers between Asset Hub and parachains.
+
+The service will support bidirectional XCM transfers between Asset Hub and selected parachains. It is not feasible to provide generic support for all Asset Hub-parachain channels due to differences in pallets used for asset management and XCM message initialization. For asset management, some parachains use FRAME `assets` pallet while others use ORML `tokens` pallet. There are also networks, like Equilibrium, who uses their own custom pallet. For XCM messages, some parachains use Polkadot `xcm` pallet while others use ORML `xtokens` pallet. To keep the project manageable, our initial focus will be on supporting Moonbeam, Astar, and Acala. In the future, we plan to expand support to include more channels and XCM protocols.
```

Comment:
I don't see what the pallets have to do with creating this. As long as you can decode the XCM programs being sent, match them across chains, and see whether the `Outcome` was successful or not, you should have a good idea as to what happened without knowledge of specific pallets. Sure, some chains use different pallets to construct XCM programs, but the programs (the XCM message format) obviously has to be compatible across the chains, otherwise they could not interact.

Reply:
We have taken your input into account, investigated deeper and updated the monitoring flow accordingly.